### PR TITLE
Add finish debugger command.

### PR DESCRIPTION
**Pull request by knightburton**
With this command the engine continue running just after the function
in the current stack frame returns.

JerryScript-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/debugger/debugger.c [hunk 1]
```diff
@@ -35,7 +35,7 @@
  * debugger versioning.
  */
 JERRY_STATIC_ASSERT (JERRY_DEBUGGER_MESSAGES_OUT_MAX_COUNT == 27
-                     && JERRY_DEBUGGER_MESSAGES_IN_MAX_COUNT == 18
+                     && JERRY_DEBUGGER_MESSAGES_IN_MAX_COUNT == 19
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.c]: Please increase the ERRY_DEBUGGER_VERSION number.

File: jerry-core/debugger/debugger.c [hunk 2]
```diff
@@ -438,6 +438,16 @@ jerry_debugger_process_message (uint8_t *recv_buffer_p, /**< pointer to the rece
       return true;
     }
 
+    case JERRY_DEBUGGER_FINISH:
+    {
+      JERRY_DEBUGGER_CHECK_PACKET_SIZE (jerry_debugger_receive_type_t);
+
+      JERRY_DEBUGGER_SET_FLAGS (JERRY_DEBUGGER_VM_STOP);
+      JERRY_CONTEXT (debugger_stop_context) = JERRY_CONTEXT (vm_top_context_p->prev_context_p);
```
Comment (zherczeg) [inline jerry-core/debugger/debugger.c]: It is worth a comment to describe the NULL case (and why this code is correct).

File: jerry-debugger/jerry-client-ws.html [hunk 3]
```diff
@@ -1367,6 +1368,7 @@ <h2>JerryScript HTML (WebSocket) Debugger Client</h2>
               "  continue|c - continue execution\n" +
               "  step|s - step-in execution\n" +
               "  next|n - execution until the next breakpoint\n" +
+              "  finish|f - continue running just after function in the current stack frame returns\n" +
```
Comment (zherczeg) [inline jerry-debugger/jerry-client-ws.html]: comment: continue running until the current function returns

File: jerry-debugger/jerry-client-ws.py [hunk 4]
```diff
@@ -286,6 +287,13 @@ def do_next(self, args):
 
     do_n = do_next
 
+    def do_finish(self, args):
+        """ Continue running just after function in the current stack frame returns """
```
Comment (zherczeg) [inline jerry-debugger/jerry-client-ws.py]: Ditto.

File: tests/debugger/do_finish.cmd [hunk 5]
```diff
@@ -0,0 +1,6 @@
+step
+step
+finish
+step
+finish
+continue
```
Comment (zherczeg) [inline tests/debugger/do_finish.cmd]: last one could be a finish as well

**Comment by knightburton** [comment]
@zherczeg I've updated the PR based on your requests.

**Comment by martijnthe** [comment]
For future changes: I think it would have been nicer to not shuffle the existing message enums.
That way it would be a bit easier for a client to support multiple versions.

**Pull request merged**